### PR TITLE
feat: fixed the issue edit pbas by orgIdentifier

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

RDCC-1641

### Change description ###

fixed the issue edit pbas by orgIdentifier

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java [hunk 1]
```diff
@@ -117,12 +117,14 @@ private Organisation saveOrganisation(Organisation organisation) {
...
     public void addPbaAccountToOrganisation(Set<String> paymentAccounts, Organisation organisation) {
         if (paymentAccounts != null) {
+            log.info("{}:: Inside  addPbaAccountToOrganisation ::{}::" + paymentAccounts);
             PaymentAccountValidator.checkPbaNumberIsValid(paymentAccounts);
 
             paymentAccounts.forEach(pbaAccount -> {
                 PaymentAccount paymentAccount = new PaymentAccount(pbaAccount.toUpperCase());
                 paymentAccount.setOrganisation(organisation);
                 PaymentAccount persistedPaymentAccount = paymentAccountRepository.save(paymentAccount);
+                log.info("{}:: paymentAccount saved in the DB ::{}::");
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java:127]: params not given in logs

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java [hunk 2]
```diff
@@ -29,19 +29,8 @@
...
-
-            } else if ("true".equalsIgnoreCase(config.getPbaFromUserAccountMap())) {
-
-                List<PaymentAccount>  userMapPaymentAccount = getPaymentAccountsFromUserAccountMap(professionalUser.getUserAccountMap());
-
-                pbaNumbers = getPbaNumbersFromPaymentAccount(userMapPaymentAccount, professionalUser.getOrganisation().getPaymentAccounts());
-
-            }
-
+            log.info("findLegacyPbaAccountByUserEmail :: inside {}:");
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java:32]: params not given i logger

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java [hunk 3]
```diff
@@ -100,37 +100,38 @@ public void retrievePaymentAccountsByPbaEmailWhenConfigTrue() {
 
         verify(organisationMock, times(1)).setPaymentAccounts(any());
         verify(organisationMock, times(1)).setUsers(any());
-        verify(organisationMock, times(1)).setPaymentAccounts(any());
         verify(professionalUserRepositoryMock, times(1)).findByEmailAddress("some-email");
     }
 
+    @Test(expected = Exception.class)
+    public void testThrowsExceptionWhenEmailInvalid() {
+        when(sut.findPaymentAccountsByEmail("some-email")).thenReturn(organisation);
```
Comment (shreedhar-hmcts) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java:108]: actual method is not called, just used when

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java [hunk 4]
```diff
@@ -100,37 +100,38 @@ public void retrievePaymentAccountsByPbaEmailWhenConfigTrue() {
...
+    }
+
     @Test
-    public void retrievePaymentAccountsByPbaEmailWhenConfigFalse() {
+    public void editPaymentAccountsByOrganisationIdentifier() {
```
Comment (shreedhar-hmcts) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java]: prepend test to method name

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImpl.java [hunk 5]
```diff
@@ -109,14 +106,16 @@ public void addPaymentAccountsToOrganisation(PbaEditRequest pbaEditRequest, Orga
...
         List<PaymentAccount> paymentAccounts = organisation.getPaymentAccounts();
         SuperUser superUser = organisation.getUsers().get(0);
+        log.info("addUserAndPaymentAccountsToUserAccountMap::Payment Accounts Size{}::" + paymentAccounts.size());
 
         /** Please note:
          * Currently only the Super User of an Organisation is linked to the Payment Accounts via the User Account Map.
          * If this changes then the below logic will need to change accordingly */
         userAccountMapService.persistedUserAccountMap(superUser.toProfessionalUser(), paymentAccounts);
-
+        log.info("{}::After UserAccountMap persisted::{}::");
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImpl.java:118]: params not given in logs

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java (hunk 2 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java:32]: wontedly not printing the params

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java (hunk 1 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java:127]: wontedly not printing the params

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImpl.java (hunk 5 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImpl.java:118]: wontedly not printing the params

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java (hunk 4 above)
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java]: added not sure why it is not reflecting ..

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java (hunk 3 above)
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java:108]: updated now

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java (hunk 1 above)
Comment (shreedhar-hmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java:127]: ok so then remove {} from everywhere

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java [hunk 6]
```diff
@@ -100,49 +100,50 @@ public void retrievePaymentAccountsByPbaEmailWhenConfigTrue() {
...
         final List<PaymentAccount> paymentAccounts = new ArrayList<>();
         paymentAccounts.add(new PaymentAccount());
         ProfessionalUser professionalUserMock = mock(ProfessionalUser.class);
 
         when(professionalUserMock.getOrganisation()).thenReturn(organisationMock);
-        when(applicationConfigurationMock.getPbaFromUserAccountMap()).thenReturn("false");
         when(organisationMock.getStatus()).thenReturn(OrganisationStatus.ACTIVE);
         when(organisationMock.getPaymentAccounts()).thenReturn(paymentAccounts);
-        when(professionalUserRepositoryMock.findByEmailAddress("some-email")).thenReturn(professionalUserMock);
+        when(organisationRepositoryMock.findByOrganisationIdentifier(any(String.class))).thenReturn(organisation);
```
Comment (AdilOoze) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java:122]: can you add a `verify` for `organisationRepositoryMock.findByOrganisationIdentifier`

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImplTest.java [hunk 7]
```diff
@@ -50,28 +43,9 @@ public void setup() {
...
-        UserAccountMapId newUserAccountMapId = new UserAccountMapId(professionalUser, paymentAccount);
-        UserAccountMap userAccountMap = new UserAccountMap(newUserAccountMapId);
-        userAccountMapData.add(userAccountMap);
-
-        paymentAccountPbaNumbers = sut.findLegacyPbaAccountByUserEmail(professionalUser);
-
-        assertThat(paymentAccountPbaNumbers).isNotNull();
-        assertThat(paymentAccountPbaNumbers.size()).isGreaterThanOrEqualTo(0);
-
-        verify(configurationMock, times(2)).getPbaFromUserAccountMap();
```
Comment (AdilOoze) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImplTest.java:75]: Any reason this test was deleted - seems like a test we should have for the Legacy endpoint?

Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImplTest.java:75]: true condition is not required at the moment.

File: src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java (hunk 6 above)
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImplTest.java:125]: ok

**Pull request merged**